### ISSUE TITLE
npm install fails with comment syntax error in libuv compilation

### ISSUE THREAD

**Issue by jkozak** [versatica/mediasoup#860]
### Your environment

- Operating system: NixOS 22.05
- Node version: 16.15.0
- npm version: 8.5.5
- gcc/clang version: 9.3.0, 10.3.0, 11.3.0
- mediasoup version: 3.9.2, 3.10.0
- mediasoup-client version: 3.6.47

### Issue description

I have tried this with various C compilers and mediasoup versions (see above), same result each time.  The offending C++ style comment has been in the libuv source for a year.

relevant bit of build output:

```
npm ERR! [971/1322] Compiling C object subprojects/openssl-3.0.2/libssl.a.p/ssl_statem_statem_lib.c.o
npm ERR! [972/1322] Compiling C object subprojects/libuv-v1.43.0/libuv.a.p/src_fs-poll.c.o
npm ERR! [973/1322] Compiling C object subprojects/libuv-v1.43.0/libuv.a.p/src_idna.c.o
npm ERR! [974/1322] Compiling C object subprojects/openssl-3.0.2/libssl.a.p/ssl_methods.c.o
npm ERR! [975/1322] Compiling C object subprojects/libuv-v1.43.0/libuv.a.p/src_random.c.o
npm ERR! [976/1322] Compiling C object subprojects/libuv-v1.43.0/libuv.a.p/src_strscpy.c.o
npm ERR! [977/1322] Compiling C object subprojects/openssl-3.0.2/libssl.a.p/ssl_s3_enc.c.o
npm ERR! [978/1322] Compiling C object subprojects/openssl-3.0.2/libssl.a.p/ssl_statem_extensions.c.o
npm ERR! [979/1322] Compiling C object subprojects/openssl-3.0.2/libssl.a.p/ssl_ssl_conf.c.o
npm ERR! [980/1322] Compiling C object subprojects/libuv-v1.43.0/libuv.a.p/src_uv-data-getter-setters.c.o
npm ERR! [981/1322] Compiling C object subprojects/openssl-3.0.2/libssl.a.p/ssl_record_ssl3_record.c.o
npm ERR! [982/1322] Compiling C object subprojects/openssl-3.0.2/libcrypto.a.p/providers_implementations_encode_decode_encode_key2any.c.o
npm ERR! [983/1322] Compiling C object subprojects/openssl-3.0.2/libssl.a.p/ssl_record_rec_layer_s3.c.o
npm ERR! [984/1322] Compiling C object subprojects/libuv-v1.43.0/libuv.a.p/src_threadpool.c.o
npm ERR! [985/1322] Compiling C object subprojects/libuv-v1.43.0/libuv.a.p/src_timer.c.o
npm ERR! [986/1322] Compiling C object subprojects/openssl-3.0.2/libssl.a.p/ssl_ssl_cert.c.o
npm ERR! [987/1322] Compiling C object subprojects/libuv-v1.43.0/libuv.a.p/src_inet.c.o
npm ERR! [988/1322] Compiling C object subprojects/openssl-3.0.2/libssl.a.p/ssl_t1_trce.c.o
npm ERR! [989/1322] Compiling C object subprojects/libuv-v1.43.0/libuv.a.p/src_version.c.o
npm ERR! [990/1322] Compiling C object subprojects/openssl-3.0.2/libssl.a.p/ssl_statem_statem_clnt.c.o
npm ERR! [991/1322] Compiling C object subprojects/libuv-v1.43.0/libuv.a.p/src_unix_dl.c.o
npm ERR! [992/1322] Compiling C object subprojects/libuv-v1.43.0/libuv.a.p/src_unix_async.c.o
npm ERR! [993/1322] Compiling C object subprojects/libuv-v1.43.0/libuv.a.p/src_unix_getnameinfo.c.o
npm ERR! [994/1322] Compiling C object subprojects/libuv-v1.43.0/libuv.a.p/src_unix_getaddrinfo.c.o
npm ERR! [995/1322] Compiling C object subprojects/openssl-3.0.2/libssl.a.p/ssl_statem_statem_srvr.c.o
npm ERR! [996/1322] Compiling C object subprojects/openssl-3.0.2/libssl.a.p/ssl_ssl_ciph.c.o
npm ERR! [997/1322] Compiling C object subprojects/openssl-3.0.2/libssl.a.p/ssl_t1_lib.c.o
npm ERR! [998/1322] Compiling C object subprojects/libuv-v1.43.0/libuv.a.p/src_unix_loop.c.o
npm ERR! [999/1322] Compiling C object subprojects/libuv-v1.43.0/libuv.a.p/src_unix_loop-watcher.c.o
npm ERR! [1000/1322] Compiling C object subprojects/openssl-3.0.2/libssl.a.p/ssl_s3_lib.c.o
npm ERR! [1001/1322] Compiling C object subprojects/libuv-v1.43.0/libuv.a.p/src_unix_random-devurandom.c.o
npm ERR! [1002/1322] Compiling C object subprojects/libuv-v1.43.0/libuv.a.p/src_unix_stream.c.o
npm ERR! FAILED: subprojects/libuv-v1.43.0/libuv.a.p/src_unix_stream.c.o
npm ERR! cc -Isubprojects/libuv-v1.43.0/libuv.a.p -Isubprojects/libuv-v1.43.0 -I../../../subprojects/libuv-v1.43.0 -I../../../subprojects/libuv-v1.43.0/include -I../../../subprojects/libuv-v1.43.0/src -fdiagnostics-color=always -DNDEBUG -std=c89 -O3 -fPIC -pthread -fno-strict-aliasing -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_GNU_SOURCE -D_POSIX_C_SOURCE=200112 -MD -MQ subprojects/libuv-v1.43.0/libuv.a.p/src_unix_stream.c.o -MF subprojects/libuv-v1.43.0/libuv.a.p/src_unix_stream.c.o.d -o subprojects/libuv-v1.43.0/libuv.a.p/src_unix_stream.c.o -c ../../../subprojects/libuv-v1.43.0/src/unix/stream.c
npm ERR! ../../../subprojects/libuv-v1.43.0/src/unix/stream.c: In function ‘uv__write’:
npm ERR! ../../../subprojects/libuv-v1.43.0/src/unix/stream.c:929:3: error: C++ style comments are not allowed in ISO C90
npm ERR!   929 |   // XXX(jwn): this must call uv__stream_flush_write_queue(stream, n) here, since we won't generate any more events
npm ERR!       |   ^
npm ERR! ../../../subprojects/libuv-v1.43.0/src/unix/stream.c:929:3: note: (this will be reported only once per input file)
```

**Comment by nazar-pc** [comment]
This was already reported and is being worked on, no need to report multiple times.

**Comment by ibc** [comment]
Guys, please, check other reported issues and the forum.

**Comment by ibc** [comment]
Fixed in 3.10.1.
